lib/src/flutter_search_bar_base.dart

```diff
@@ -132,7 +136,18 @@ class SearchBar {
   ///
   AppBar buildSearchBar(BuildContext context) {
     ThemeData theme = Theme.of(context);
-    Color buttonColor = inBar ? null : theme.iconTheme.color;
+    TextStyle hintStyle = TextStyle(
+      color: inBar
+          ? theme.appBarTheme?.textTheme?.headline4?.color ?? theme.primaryTextTheme?.headline4?.color
+          : theme.textTheme.headline4.color,
+    );
+    TextStyle inputStyle = inBar ? theme.appBarTheme?.textTheme?.subtitle1 ?? theme.primaryTextTheme?.subtitle1 : null;
```

Comment:
Is there any particular reason we're using subtitle1 for the style here? It seems slightly arbitrary, but then again so does headline4 (which has been in use here for a long time). My best guess is that subtitle1 is intended to be used for its color since subtitles are generally a more muted version of the headline color, but why are we also using the weight + size + case + spacing (all of which are a part of TextStyle) instead of just grabbing the color?

Reply:
> Is there any particular reason we're using subtitle1

Same as for the hardcoded disabled colors, this is the default for `TextField` [in the Flutter definition](https://github.com/flutter/flutter/blob/fb508db94d9ecfac3b21f961bfd835f77eca9b88/packages/flutter/lib/src/material/text_field.dart#L1260):

```
final TextStyle style = theme.textTheme.subtitle1.merge(widget.style);
```

All other properties are merged anyway, and if you pass them with `theme.appBarTheme?.textTheme?.subtitle1`, you can at least customize it when needed. Plus, because you're in the app bar, you may want a proper different style, which matches the whole content of the app bar.

An improvement to all of this would be to have a "search bar theme".

Reply:
Weird. I wish there was less random hardcoded color stuff and more information on when to use each color/style, but oh well. I do agree there should be a "search bar theme", but I am still torn on whether this module is really useful enough on its own to justify making breaking changes like that (especially now that we have SearchContext, even if it isn't quite as fluid)